Tests/SwiftDocCTests/Rendering/SampleDownloadTests.swift

```diff
@@ -240,4 +240,22 @@ class SampleDownloadTests: XCTestCase {
         let reference = try XCTUnwrap(renderNode.references[identifier.identifier])
         XCTAssert(reference is ExternalLocationReference)
     }
+
+    /// Ensure that a DownloadReference where the URL is different from the reference identifier
+    /// can still round-trip through an ExternalLocationReference with the URL and reference identifier intact.
+    func testRoundTripWithDifferentUrl() throws {
```

Comment:
Could we add a full test fixture for this too? I like this as a targeted test but I think it would be good to include a RenderJSON test fixture where we've manually modified the URL property and ensure it round-trips as expected. Just to be extra-safe.